### PR TITLE
dev/core#2000 Move afform into core extensions

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Moving afform into core means we can start to recreate core forms as afforms, in particular where
they are problematic and / or hard to maintain.

Before
----------------------------------------
Maintained as separate repo

After
----------------------------------------
Maintained as part of core repo

Technical Details
----------------------------------------
Per https://lab.civicrm.org/dev/core/-/issues/2000 my understanding of the scope of this round of work is just to move it into core & wire it up. Future rounds of work will transition it from 'present' to installed - probably with installed happening first on buildkit installs. There are some technical issues for later rounds around the scenario where the extension is present in the core directory and in the extensions directory

Comments
----------------------------------------
We need to continue to have a lighter review requirement for afform while it's still not enabled on install

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
I re-ran civix & added a commit & also here https://lab.civicrm.org/extensions/afform/-/merge_requests/28/diffs

**Comment by totten** [comment]
When we merge, I'd like to preserve history if we can, eg http://gbayer.com/development/moving-files-from-one-git-repository-to-another-preserving-history/

**Comment by eileenmcnaughton** [comment]
@totten hmm OK - did we do that for any of the others?

**Comment by totten** [comment]
Well, we haven't exactly done this a lot. :) For comparison:

* civicrm-core's svn to git: We tried to migrate the history but couldn't get it to work. (IIRC, the migration scripts were crashing when trying to deal with the full history. The svn repo had a long history and an atypical topology.)
* api4: For this migration, it was de-constructed so that it would not longer be an extension. I don't think we were clever enough to do the de-construction and also figure out history-migration at the same time.
* flexmailer: No, though I feel like we could've done that better (because it's a simpler case). My guess is that the previous cases looked precedents.

Basically, if we have a situation where it is feasible to keep the history, then I think that's better.

(I'm experimenting here: https://gist.github.com/totten/5881476fdea542a57d60815da37ea3cf )

**Comment by totten** [comment]
@eileenmcnaughton I've updated the migration script at https://gist.github.com/totten/5881476fdea542a57d60815da37ea3cf -- it gives a branch in which:

* The base is `civicrm-core.git@master`
* The folder `ext/afform` contains the content from `afform.git`
* The full history is reproduced for files in `ext/afform`.

Example output:

https://github.com/civicrm/civicrm-core/compare/master...totten:master-afform?expand=1

If the idea is to continue patching `afform.git` for the moment, then we can just re-run it when that's done.

**Comment by eileenmcnaughton** [comment]
@totten @seamuslee001 not sure about the last 2 style warnings

**Comment by seamuslee001** [comment]
@eileenmcnaughton add in lines like https://github.com/civicrm/civicrm-core/blob/master/ext/flexmailer/tests/phpunit/bootstrap.php#L14 around the eval

**Comment by eileenmcnaughton** [comment]
@totten with this merged to afform https://lab.civicrm.org/extensions/afform/-/merge_requests/30 style is solved - I think the next step is to close this & for you to do a replacement?

**Comment by eileenmcnaughton** [comment]
@totten I see you just merged the PR into afform - what is the next step? Should I close this & you can do some magic?

**Comment by totten** [comment]
@eileenmcnaughton I've re-run the [migration script](https://gist.github.com/totten/5881476fdea542a57d60815da37ea3cf) and force-pushed to your branch. So this PR now has a full history.

**Comment by eileenmcnaughton** [comment]
@seamuslee @coleman afform PR is now passing. It was in last week's dev digest but I think one of you should merge as it's technically my PR but all the commits were added by @totten

**Comment by colemanw** [comment]
Merged. I also archived the gitlab repo to avoid confusion.

**Comment by eileenmcnaughton** [comment]
Cool - so test wiring just needs doing

**Comment by colemanw** [comment]
@totten can you pls enable the afform tests on core CI?
